internal/acceptance/lakeview_dashboard_test.go

```diff
@@ -0,0 +1,481 @@
+	`, template.EmbedCredentials)
+	}
+	templateString += `}
+	`
+	return templateString
+}
+
+// Altough EmbedCredentials is an optional field, please specify its value if you want to modify it.
+func (t *TemplateStruct) SetAttributes(mapper map[string]string) TemplateStruct {
+	// Switch case for each attribute. If it is set in the mapper, set it in the struct
+	if val, ok := mapper["display_name"]; ok {
+		t.DisplayName = val
+	}
+	if val, ok := mapper["warehouse_id"]; ok {
+		t.WarehouseId = val
+	}
+	if val, ok := mapper["parent_path"]; ok {
+		t.ParentPath = val
+	}
+	if val, ok := mapper["serialized_dashboard"]; ok {
+		t.SerializedDashboard = val
+	}
+	if val, ok := mapper["file_path"]; ok {
+		t.FilePath = val
+	}
+	if val, ok := mapper["embed_credentials"]; ok {
+		if val == "true" {
+			t.EmbedCredentials = ""
+		} else {
+			t.EmbedCredentials = val
+		}
+	}
+	return *t
+}
+
+func TestAccLakeviewDashboard(t *testing.T) {
+	var template TemplateStruct
+	workspaceLevel(t, step{
+		Template: MakeTemplate(template.SetAttributes(map[string]string{
+			"display_name":         "Monthly Traffic Report",
+			"warehouse_id":         "{env.TEST_DEFAULT_WAREHOUSE_ID}",
+			"parent_path":          "/Shared/provider-test",
+			"serialized_dashboard": `{\"pages\":[{\"name\":\"new_name\",\"displayName\":\"New Page\"}]}`,
+		})),
+		Check: resourceCheck("databricks_lakeview_dashboard.d1", func(ctx context.Context, client *common.DatabricksClient, id string) error {
+			w, err := client.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			dashboard, err := w.Lakeview.Get(ctx, dashboards.GetDashboardRequest{
+				DashboardId: id,
+			})
+			if err != nil {
+				return err
+			}
+			assert.Equal(t, "Monthly Traffic Report", dashboard.DisplayName)
+			fmt.Println(dashboard.SerializedDashboard)
+			require.NoError(t, err)
+			return nil
+		}),
+	})
+}
+
+func TestAccDashboardWithSerializedJSON(t *testing.T) {
+	var template TemplateStruct
+	workspaceLevel(t, step{
+		Template: MakeTemplate(template.SetAttributes(map[string]string{
+			"display_name":         "Monthly Traffic Report",
+			"warehouse_id":         "{env.TEST_DEFAULT_WAREHOUSE_ID}",
+			"parent_path":          "/Shared/provider-test",
+			"serialized_dashboard": `{\"pages\":[{\"name\":\"new_name\",\"displayName\":\"New Page\"}]}`,
+			"embed_credentials":    "false",
+		})),
+		Check: resourceCheck("databricks_lakeview_dashboard.d1", func(ctx context.Context, client *common.DatabricksClient, id string) error {
+			w, err := client.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			dashboard, err := w.Lakeview.Get(ctx, dashboards.GetDashboardRequest{
+				DashboardId: id,
+			})
+			if err != nil {
+				return err
+			}
+			assert.Equal(t, "Monthly Traffic Report", dashboard.DisplayName)
+			fmt.Println(dashboard.SerializedDashboard)
+			require.NoError(t, err)
+			return nil
+		}),
+	}, step{
+		Template: MakeTemplate(template.SetAttributes(map[string]string{
```

Comment:
The fact that we keep updating the template, but reuse some attributes seems a bit off to me. I would prefer to see here all attributes for the current step. @edwardfeng-db what do you think?

Reply:
Would not that be same as writing the full template. I thing the reasoning that Edward gave was that in tests with many steps if we change only one attribute then why to write the full template. Also, the change would be more visible if we just mention that.

Reply:
It's a bit less than the whole template (4 lines vs ~10). And mappings have better syntax highlighting. But yes, we don't want to make it too big.

Reply:
I'd say I would still advocate for having a helper methods, but if you have strong opinion against it, don't have to block on me.

I'll leave the decision to @Divyansh-db

Reply:
I like having helper methods. I just found it a bit hard to read. But it could be because the git view is not as great as the vscode one. Let's go with this. We can change it later.